testng-core/src/test/kotlin/test/SimpleBaseTest.kt

```diff
@@ -0,0 +1,488 @@
+package test
+
+import org.assertj.core.api.Assertions
+import org.testng.*
+import org.testng.annotations.ITestAnnotation
+import org.testng.collections.Lists
+import org.testng.internal.annotations.AnnotationHelper
+import org.testng.internal.annotations.DefaultAnnotationTransformer
+import org.testng.internal.annotations.IAnnotationFinder
+import org.testng.internal.annotations.JDK15AnnotationFinder
+import org.testng.xml.*
+import org.testng.xml.internal.Parser
+import java.io.*
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+import kotlin.collections.ArrayList
+
+open class SimpleBaseTest {
+
+    companion object {
+        private const val TEST_RESOURCES_DIR = "test.resources.dir"
+
+        @JvmStatic
+        fun run(vararg testClasses: Class<*>) : InvokedMethodNameListener {
+            return run(false, *testClasses)
+        }
+
+        @JvmStatic
+        private fun run(skipConfiguration: Boolean, tng: TestNG) =
+            InvokedMethodNameListener(skipConfiguration).apply {
+            tng.addListener(this)
+            tng.run()
+            return this
+        }
+
+        @JvmStatic
+        fun run(skipConfiguration: Boolean, vararg testClasses: Class<*>)
+                : InvokedMethodNameListener {
+            val tng = create(*testClasses)
+            return run(skipConfiguration, tng)
+        }
+
+        @JvmStatic
+        fun create(vararg testClasses: Class<*>): TestNG {
+            val result = create()
+            result.setTestClasses(testClasses)
+            return result
+        }
+
+        @JvmStatic
+        fun create(): TestNG {
+            val result = TestNG()
+            result.setUseDefaultListeners(false)
+            return result
+        }
+
+        @JvmStatic
+        fun create(xmlSuite: XmlSuite): TestNG {
+            val result = create()
+            result.setXmlSuites(listOf(xmlSuite))
+            return result
+        }
+
+        @JvmStatic
+        fun run(vararg suites: XmlSuite): InvokedMethodNameListener {
+            return run(false, *suites)
+        }
+
+        @JvmStatic
+        fun run(skipConfiguration: Boolean, vararg suites: XmlSuite): InvokedMethodNameListener {
+            val tng = create(*suites)
+            return run(skipConfiguration, tng)
+        }
+
+        @JvmStatic
+        protected fun create(outputDir: Path, vararg testClasses: Class<*>): TestNG {
+            val result = create(*testClasses)
+            result.outputDirectory = outputDir.toAbsolutePath().toString()
+            return result
+        }
+
+        @JvmStatic
+        protected fun create(vararg suites: XmlSuite): TestNG {
+            return create(listOf(*suites))
+        }
```

Comment:
Ditto.

Most of these `return` functions can be simplified by a `fun foo()... = ...`

Reply:
fixed